### DISCUSSION TITLE
[Feature] save searches as albums

### DISCUSSION THREAD

**Discussion by cunicode** [immich-app/immich#8331] [Feature Request]
### The feature

As we have the semantic search, it could be useful to save (some) searches as albums (smart albums?)

use case: 
- I search for "beach"
- I decide to save that search as album
- that album will contain existing and future assets that match "beach" 

implementation:
since searches already create urls, an easy way to implement this could be to storing the links only (without album features of adding)

(similar to this feature [https://github.com/immich-app/immich/discussions/1673](https://github.com/immich-app/immich/discussions/1673) but on demand after search is performed)

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by cunicode** [comment]
On the mobile app, the categories albums "screenshots" and "selfies" seem to work like that, as the results are the same as searching for "screenshot" or "selfie"

**Comment by m3brown** [comment]
I :+1: this feature request!

A driving factor is I'd **really** love to give my (authenticated) parents access to all photos of my kids, auto-updating over time.

Some thoughts:

1. It's worth emphasizing rather than assuming folks understand: I think the primary benefit of saved searches is the dynamic nature of the content. Maintaining albums takes time and energy if new photos are to be added over time.

1. I'm not familiar with the Immich codebase, but it could be that the implementation logistics for an Album are not compatible with search results. I'm thinking access control logic. In my mind, this feature does not need to be an "Album" -- it could be something like "Saved Searches" and still provide value alongside the existing Album functionality.

1. For saved searches to be truly valuable for me, I'd want to share them with others.
    - For example, I'd like to save the date ranges for all my family vacations so my spouse can easily view them.
    - Anonymous links could be nice, but I'm thinking "Shared Searches" (or Albums) in the UI for authenticated users is just as good.

**Reply by cunicode** [reply]
Saving urls as buttons could do the trick.... 
http://127.0.0.1:2283/search?query=%7B%22query%22%3A%22forest%22%7D

**Reply by m3brown** [reply]
Yes! This reminds me of Jira's feature called Quick Filters

https://support.atlassian.com/jira-software-cloud/docs/configure-quick-filters/

Their search filters could also be a useful influence, it has the concept of sharing with others.

https://confluence.atlassian.com/jirasoftwareserver0818/saving-your-search-as-a-filter-1082527265.html

**Comment by ThreeAurora** [comment]
Very creative feature, I thought about it, and if it did come to fruition, I wouldn't be able to live without it

**Comment by cunicode** [comment]
my current workaround is to save bookmarks on the browser: 
http://localhost:2283/search?query=%7B"query"%3A"red"%7D 

but since I mostly use immich on the phone app, it is not very useful :(

**Comment by slanglade** [comment]
I totally second this feature request !

For me, it's really the last thing that keep me from ditching Synology Photo. To be able to define an album using a search would be really useful. 
You can create an album dynamically compiling all photos with specific people in it (read: family or friends album) and/or base on geolocalisation and/or with contextual searches (if activated on your instance) etc. and then share it. It will automagically be updated with all new photos according to filter.

That's a hell of a usefull feature, and even if the current structuration of what is technically an album in Immich can be a tricky problem (I don't know), it will really be worth the effort.

**Reply by slanglade** [reply]
And don't forget the fact that by giving a user a link to this shared album, it also save you the hassle of defining precise access rights on each and every corresponding photos. Which is also a big winning point.

**Comment by tylers-username** [comment]
This would lay the foundation for automated albums such as "Tyler, in 2024" as well, right?

**Reply by hasanm3** [reply]
Earlier today I submitted a Feature-request for 'dynamic albums', whereby Albums should be able to accept a 'criteria' and all media matching that criteria should be included to the album. The criteria could be tags, people, timeline or search queries, or a combination of these.

**Comment by Ged296123** [comment]
I believe this is tracked in #3641

**Comment by gregstah** [comment]
Here's my workaround, unfortunately, the app doesn't support tags so, there's that. 

You can assign new tags to any image, click info, then **Tags->Add** if the Tag doesn't exist, Type  it in and hit **Enter** It'll create the Tag and you can assign it. Like I said, desktop only.

**Comment by LnLcFlx** [comment]
I think one of the two approaches would be best:
1. Be able to save a search/filter as a reusable timeline view besides the default timeline (e.g. like the views in [Paperless NGX](https://docs.paperless-ngx.com/#features)) (more dynamic)
2. Combine nested albums #2073 and rule based automation #1673 on the roadmap: An album could be used as a reusable search view and using workflows it can be populated based on rules. (less dynamic)

The original proposal is more like 2.
If the goal is to be able to convert a quick search into a permanent rule, one could just add a button "copy search as rule" or similar and then "paste" somehow in the workflow feature.
Or add a button "create album for search and add corresponding workflow".

**Reply by bo0tzz** [reply]
In practice, a button to turn the current search into a workflow is probably what the implementation will end up looking like yes.
